tests/test_Summary.cpp

```diff
@@ -0,0 +1,365 @@
+
+    BOOST_CHECK_CLOSE( 30.0 + 30.1, ecl_sum_get_well_var( resp, 0, "W_3", "WLPR" ), 1e-5 );
+    BOOST_CHECK_CLOSE( (30.0 + 30.1) / day, ecl_sum_get_well_var( resp, 0, "W_3", "WLPT" ), 1e-5 );
+    BOOST_CHECK_CLOSE( (30.0 + 30.1), ecl_sum_get_well_var( resp, 1, "W_3", "WLPR" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 2 * (30.0 + 30.1) / day, ecl_sum_get_well_var( resp, 1, "W_3", "WLPT" ), 1e-5 );
+
+    rmfiles( fname );
+}
+
+BOOST_AUTO_TEST_CASE(W_WOG_PRH) {
+    const auto deck = Parser().parseFile( path, ParseContext() );
+    EclipseState es( deck, ParseContext() );
+    Summary summary( *deck, es );
+    const auto wells = result_wells();
+
+    std::string fname = "sum_test_W_WOG_PRH";
+
+    out::Summary writer( es, summary, fname );
+    writer.add_timestep( 1, 1, es, wells );
+    writer.write();
+
+    auto res = readsum( fname );
+    const auto* resp = res.get();
+
+    BOOST_CHECK_CLOSE( 10, ecl_sum_get_well_var( resp, 0, "W_1", "WWPRH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 20, ecl_sum_get_well_var( resp, 0, "W_2", "WWPRH" ), 1e-5 );
+
+    BOOST_CHECK_CLOSE( 10.1, ecl_sum_get_well_var( resp, 0, "W_1", "WOPRH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 20.1, ecl_sum_get_well_var( resp, 0, "W_2", "WOPRH" ), 1e-5 );
+
+    BOOST_CHECK_CLOSE( 10.2, ecl_sum_get_well_var( resp, 0, "W_1", "WGPRH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 20.2, ecl_sum_get_well_var( resp, 0, "W_2", "WGPRH" ), 1e-5 );
+
+    rmfiles( fname );
+}
+
+BOOST_AUTO_TEST_CASE(W_WOG_PTH) {
+    const auto deck = Parser().parseFile( path, ParseContext() );
+    EclipseState es( deck, ParseContext() );
+    Summary summary( *deck, es );
+    const auto wells = result_wells();
+
+    std::string fname = "sum_test_W_WOG_PRH";
+
+    out::Summary writer( es, summary, fname );
+    writer.add_timestep( 1, 1, es, wells );
+    writer.add_timestep( 2, 1, es, wells );
+    writer.write();
+
+    auto res = readsum( fname );
+    const auto* resp = res.get();
+
+    BOOST_CHECK_CLOSE( 2 * 10.0 / day, ecl_sum_get_well_var( resp, 1, "W_1", "WWPTH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 2 * 20.0 / day, ecl_sum_get_well_var( resp, 1, "W_2", "WWPTH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 2 * 10.1 / day, ecl_sum_get_well_var( resp, 1, "W_1", "WOPTH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 2 * 20.1 / day, ecl_sum_get_well_var( resp, 1, "W_2", "WOPTH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 2 * 10.2 / day, ecl_sum_get_well_var( resp, 1, "W_1", "WGPTH" ), 1e-5 );
+    BOOST_CHECK_CLOSE( 2 * 20.2 / day, ecl_sum_get_well_var( resp, 1, "W_2", "WGPTH" ), 1e-5 );
+
+    rmfiles( fname );
+}
+
+BOOST_AUTO_TEST_CASE(WRITE) {
+    const auto deck = Parser().parseFile( path, ParseContext() );
+    EclipseState es( deck, ParseContext() );
+    Summary summary( *deck, es );
+    const auto wells = result_wells();
+
+    out::Summary writer( es, summary );
+    writer.add_timestep( 1, 1, es, wells );
+    writer.add_timestep( 2, 1, es, wells );
+    writer.write();
+}
```

Comment:
The loads are already tested (by necessity) via the other, keyword-based tests. Perhaps poorly named, the `WRITE` test is mostly there so that there is a reference output file from the section that can be inspected with other tools, such as the `ert` binaries, for non-automated testing and verification.